src/@core/application/createClientUseCase.ts

```diff
@@ -0,0 +1,64 @@
+import { ICreateClient } from '../domain/dtos/ClientDTO';
+import { IUserOutputRelations } from '../domain/dtos/UserDTO';
+import { Address } from '../domain/entities/address.entity';
+import { Client } from '../domain/entities/client.entity';
+import { User } from '../domain/entities/user.entity';
+import { IAddressRepository } from '../domain/repositories/IAddressRepository';
+import { IAdminRepository } from '../domain/repositories/IAdminRepository';
+import { IClientRepository } from '../domain/repositories/IClientRepository';
+import { IHashRepository } from '../domain/repositories/IHashRepository';
+import { IUserRepository } from '../domain/repositories/IUserRepository';
+
+export class CreateClientUseCase {
+  constructor(
+    private clientRepository: IClientRepository,
+    private userRepository: IUserRepository,
+    private addressRepository: IAddressRepository,
+    private hashRepository: IHashRepository,
+    private adminRepository: IAdminRepository,
+  ) {}
+
+  async execute(input: ICreateClient): Promise<IUserOutputRelations> {
+    // ETAPA 0: VERIFICAR SE USUÁRIO JA EXISTE
+    const findEmail = await this.adminRepository.findByEmail(input.email, true);
```

Comment:
Fica dando que não consegue ler a propriedade de undefined (TypeError: Cannot read properties of undefined (reading 'findByEmail'))